### ISSUE TITLE
Redefine `data_strategy()` parameters

### ISSUE THREAD

**Issue by clarabicalho** [macartan/gbiqq#21]
Make syntax consistent with other funs.
Define parameters in a way that makes relationships between `vars`, `probs` and `subsets` very clear especially as vectors increase in size.

**Comment by clarabicalho** [comment]
The commit above (PR #26) tries to make the strategy more clear to user by printing data strategy (could also attach as attribute for later call).

```
model <- make_model("X -> M -> Y")
model <- set_parameters(model, type = "flat")
data_strategy(model,
  n_obs = 8,
  n = NULL,
  vars = list(c("X", "Y"), "M"),
  probs = list(1, .5),
  subsets = list(NULL, "X==1 & Y==0"))
```
>Step 1: Observe X, Y (prob = 1).
Step 2: Observe M (prob = 0.5), given X==1 & Y==0.
  X  M Y
1 0 NA 0
2 1  1 0
3 1 NA 0
4 0 NA 1
5 0 NA 1
6 1 NA 1
7 0 NA 1
8 1 NA 1

IMO providing the arguments in named list (such that each list element refers to a given step) form might be more complicated (e.g., `data_strategy(..., step1 = list(vars = c("X", "Y"), prob = 1, ...), step2 = list(vars = c("M"), prob = 1, ...)`, but at the expensive of making pckg more verbose. Thoughts?

**Comment by macartan** [comment]
some issue here; I think m and n are not used consistently? 

```
Note: possible error in 'observe(complete_data, ': unused argument (n = n[[j]]) at draw_data.R:299 
Note: possible error in 'data_strategy(model, ': unused argument (ms = data_strat$ms) at gbiqq_designer.R:35 
```

**Comment by clarabicalho** [comment]
Closes with PR #26 by implementing verbose output above. @macartan @lilymedina feel free to reopen if you disagree.
